transport/server.hh

```diff
@@ -212,7 +212,6 @@ private:
         client_data make_client_data() const;
         const service::client_state& get_client_state() const { return _client_state; }
     private:
-        const ::timeout_config& timeout_config() const { return _server.timeout_config(); }
```

Comment:
nit: That's not `cql_server::timeout_config`, but `cql_server::connection::timeout_config`. The former is in fact used and you patch it to be const with https://github.com/scylladb/scylladb/pull/12531/commits/a1872332fe77475bdd06f897339ac59b4114af07